File: Tests/SourceTests.cs
using System;
using System.Text.Patterns;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Tests {
	[TestClass]
	public class SourceTests {

		[TestMethod]
		public void Peek() {
			Source Source = new Source("Hello");
			Assert.AreEqual(5, Source.Length);
			Assert.AreEqual("H", Source.Peek(1).ToString());
			Assert.AreEqual(5, Source.Length);
			Assert.AreEqual("He", Source.Peek(2).ToString());
			Assert.AreEqual(5, Source.Length);
		}

		[TestMethod]
		public void Read() {
			Source Source = new Source("Hello");
			Assert.AreEqual(5, Source.Length);
			Assert.AreEqual("H", Source.Read(1).ToString());
			Assert.AreEqual(4, Source.Length);
			Assert.AreEqual("el", Source.Read(2).ToString());
			Assert.AreEqual(2, Source.Length);
		}

	}
}

File: Tests/Patterns/SourceTests.fs
namespace Tests.Patterns

open System;
open Stringier.Patterns;
open Xunit;

module SourceTests =
    [<Fact>]
    let ``Construct from String`` () =
        let source = new Source("hello")
        Assert.Equal("hello", source.ToString())

    [<Fact>]
    let ``Construct from String, preconvert`` () =
        let source = new Source("hello", true)
        Assert.Equal("HELLO", source.ToString())